src/rules/maxClassesPerFileRule.ts

```diff
@@ -47,48 +48,32 @@ export class Rule extends Lint.Rules.AbstractRule {
     };
     /* tslint:enable:object-literal-sort-keys */
 
-    public static FAILURE_STRING_FACTORY = (maxCount: number): string => {
+    public static FAILURE_STRING(maxCount: number): string {
         const maxClassWord = maxCount === 1 ? "class per file is" : "classes per file are";
-        return `A maximum of ${maxCount} ${maxClassWord} allowed`;
+        return `A maximum of ${maxCount} ${maxClassWord} allowed.`;
     }
 
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        return this.applyWithWalker(new MaxClassesPerFileWalker(sourceFile, this.getOptions()));
+        const argument = this.ruleArguments[0];
+        const maxClasses = isNaN(argument) || (argument as number) < 0 ? 1 : argument;
```

Comment:
There's a subtle bug here. The old implementation defaults 0 to 1. The new one doesn't.
You can also simplify this line.
```ts
const maxClasses = argument > 0 ? argument : 1;
```